src/treewalk.coffee

```diff
@@ -175,7 +182,9 @@ exports.createTreewalkParser = (parse, config, root) ->
                 break
               else unless i is 0
                 end = child.bounds.start
-                if @lines[end.line][...end.column].trim().length is 0
+
+                # @lines[end.line] was breaking in python mode (index out of range)
+                if @lines[end.line-1][...end.column].trim().length is 0
```

Comment:
C mode needs it to be `@lines[end.line]`. The purpose is for cases like
```
if (a == b) {
  //whatever
}
```
The indent ends on the closing brace `}`, but we want it to end at the end of the previous line (`//whatever`), so we move it back by one.

Instead of changing to `@lines[end.line - 1]`, could you do a check like `if end.line < @lines and @lines[end.line`...etc? What are some program examples of where this breaks in python mode?